assets/js/util/tracking/createInitializeSnippet.js

```diff
@@ -53,7 +53,7 @@ export default function createInitializeSnippet( config, dataLayerTarget ) {
 			groups: 'site_kit',
 			send_page_view: config.isSiteKitScreen,
 			dimension1: referenceSiteURL,
-			dimension2: userRoles.join( ',' ),
+			dimension2: userRoles?.length ? userRoles.join( ',' ) : '',
```

Comment:
It would be a bit cleaner to fix `userRoles` once above rather than duplicate it here and below.